### ISSUE TITLE
[Stacks 2.1] Node crashing during epoch 2.1 transition

### ISSUE THREAD

**Issue by lgalabru** [stacks-network/stacks-blockchain#3489]
I have a node crashing during the epoch 2.1 transition - it looks like we're hitting a code path marked as `unreachable`:

```rust
    fn admits_type_v2_1(&self, other: &TypeSignature) -> Result<bool> {
            ...
            NoType => Err(CheckErrors::CouldNotDetermineType),
            TraitReferenceType(_) => {
                unreachable!("TraitReferenceType should not be used in epoch v2.1")
            }
        }
    }
```

Note: when the node is trying to assemble this block, my mempool contains both clarity 1 and clarity 2 contracts.

[Full debug log](https://github.com/stacks-network/stacks-blockchain/files/10405742/debg-trace.txt)

**Comment by obycode** [comment]
I'm on it. Thanks Ludo.

**Comment by jcnelson** [comment]
From the blockchain meeting:

Add an internal error type to capture all cases that are currently deemed unreachable, so at least the network doesn't crash each time we find something like this.

**Comment by jcnelson** [comment]
This was fixed in #3490.  The internal error type will be a separate issue.
